src/runtime/HalideBuffer.h

```diff
@@ -2219,7 +2223,7 @@ class Buffer {
             }
             if (flat) {
                 t[i - 1].extent *= t[i].extent;
-                for (int j = i; j < d; j++) {
+                for (int j = i; j < d - 1; j++) {
```

Comment:
t[d-1].extent will be set on line 2231, what is about .stride?

Reply:
No, we want to leave that alone -- @abadams to confirm.

Reply:
Do we still need to set the extent, given that we're not going to iterate over that dimension?

Reply:
Oooh, good point. We can probably elide that now. Let me do some hackery to verify.